experimental/packages/otlp-exporter-base/src/platform/browser/util.ts

```diff
@@ -161,3 +161,101 @@ export function sendWithXhr(
 
   sendWithRetry();
 }
+
+/**
+ * function to send metrics/spans using browser fetch
+ *     used when navigator.sendBeacon and XMLHttpRequest are not available
+ * @param body
+ * @param url
+ * @param headers
+ * @param onSuccess
+ * @param onError
+ */
+export function sendWithFetch(
+  body: string,
+  url: string,
+  headers: Record<string, string>,
+  exporterTimeout: number,
+  onSuccess: () => void,
+  onError: (error: OTLPExporterError) => void
+): void {
+  const controller = new AbortController();
+  let cancelRetry: ((e: OTLPExporterError) => void) | undefined;
+  const exporterTimer = setTimeout(() => {
+    controller.abort();
+    cancelRetry?.(new OTLPExporterError('Request Timeout'));
+  }, exporterTimeout);
+
+  const fetchWithRetry = (
+    retries = DEFAULT_EXPORT_MAX_ATTEMPTS,
+    minDelay = DEFAULT_EXPORT_INITIAL_BACKOFF
+  ) => {
+    return fetch(url, {
+      method: 'POST',
+      headers: {
+        ...defaultHeaders,
+        ...headers,
+      },
+      signal: controller.signal,
+      body,
+    }).then(
+      response => {
+        if (response.ok) {
+          response.text().then(
```

Comment:
This is not handling the "error" case where the resulting response could not be converted to text, which would result in an unhandled rejection exception event.

Also means that the `onSuccess` or `onError` won't get called.

Reply:
ie. This won't `throw` and fail the `then` that it's currently running in

Reply:
I see.
I removed converting text and diag output.
Now just the HTTP status code will only be checked the same as xhr.

Reply:
Um, it's still here...

Reply:
Ouch, sorry.
I forgot to commit the file.
fixed in 975f9f985d715f85fc3f0f75b76392e4afb03b99